.github/workflows/documentation-commit.yml

```diff
@@ -1,44 +1,35 @@
-# This workflow will install Python dependencies, run tests and lint with a
-# single version of Python. For more information see:
-# https://help.github.com/actions/language-and-framework-guides\
-# /using-python-with-github-actions
+# On each push in 15.0 branch,
+# AND if the coverage file changed,
+# build documentation branch and commit the changes
+# so that the changes are visible on the website
+# https://oca.github.io/OpenUpgrade/
 
 name: Build and commit documentation
 
 on:
   push:
-    branches: ["15.0"]
+    paths: ["docsource/modules140-150.rst"]
 
 jobs:
-  documentation:
-    runs-on: ubuntu-22.04
-    env:
-      DB: "openupgrade"
-      DB_HOST: "localhost"
-      DB_PASSWORD: "odoo"
-      DB_PORT: 5432
-      DB_USERNAME: "odoo"
-      DOWNLOADS: https://github.com/OCA/OpenUpgrade/releases/download/databases
-      ODOO: "./odoo/odoo-bin"
-      PGHOST: "localhost"
-      PGPASSWORD: "odoo"
-      PGUSER: "odoo"
+  documentation-commit:
+    runs-on: ubuntu-latest
     steps:
+      - name: Check out OpenUpgrade Documentation
+        uses: actions/checkout@v2
+        with:
+          ref: "documentation"
+
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
-          python-version: 3.7
+          python-version: '3.10'
       - name: Check out Odoo
         uses: actions/checkout@v2
         with:
           repository: odoo/odoo
-          ref: "15.0"
+          ref: "16.0"
```

Comment:
Incorrect branch

Reply:
Don't think so. here we have to install odoo to have the possibillity to install openupgradelib to have the possibility to build docs (that includes openupgradelib docstring). 
odoo version doesn't matter here. so all this part is just a copy / past of https://github.com/OCA/OpenUpgrade/blob/documentation/.github/workflows/documentation-commit.yml#L19-L57.

I fear that if we set various version of python, we get various version of sphynx, and so not the same result, if a push is done against V16 (python3.10) or V14 (python3.7).

Note : this second commit (that add again documentation-commit.yml) file on v15 is a proposal.

- In old design, if a push was done on V16, documentation was updated.
- In new design, I propose to make the update if a coverage changed in one of the 3 latest versions (v14 / v15 / v16)

Reply:
But it means that with each new branch, we have to update all the "supported" branches to match the same Odoo and libraries version... Isn't possible to "download" the stuff from the default branch instead?

Reply:
Indeed. It is far from ideal. 

For the time being, I removed this commit that readd a documentation-commit file for 14.0 and 15.0 branch. So there is no regression regarding previous design. (documentation is updated if push on 16.0 branch) 

In the near future, I would try to play with https://docs.github.com/en/actions/using-workflows/reusing-workflows, but it is not the priority.

In the meantime, I think  The resolution of this issue will make all these questions obsolete. (https://github.com/OCA/OpenUpgrade/issues/3862)